src/server/common/helpers/redis/redis-client.js

```diff
@@ -1,45 +1,68 @@
-import IoRedis from 'ioredis'
+import { Cluster, Redis } from 'ioredis'
 
-import { createLogger } from '~/src/server/common/helpers/logging/logger'
+import { createLogger } from '~/src/server/common/helpers/logging/logger.js'
+
+/**
+ * @typedef {object} RedisConfig
+ * @property {string} host
+ * @property {string} username
+ * @property {string} password
+ * @property {string} keyPrefix
+ * @property {boolean} useSingleInstanceCache
+ */
 
 /**
  * Setup Redis and provide a redis client
  *
  * Local development - 1 Redis instance
- * Out in the wild - Elasticache / Redis Cluster with username and password
- *
+ * Environments - Elasticache / Redis Cluster with username and password
+ * @param {RedisConfig} redisConfig - Redis config
  * @returns {Cluster | Redis}
  */
-function buildRedisClient(redisConfig) {
+export function buildRedisClient(redisConfig) {
   const logger = createLogger()
   const port = 6379
   const db = 0
   const keyPrefix = redisConfig.keyPrefix
   const host = redisConfig.host
+  let redisClient
 
-  const client = redisConfig.useSingleInstanceCache
```

Comment:
Think this should be:
```  
const client = redisConfig.useSingleInstanceCache
    ? new Redis({ port, host, db, keyPrefix })
    : new Cluster([{ host, port }], {
        keyPrefix,
        slotsRefreshTimeout: 10000,
        slotsRefreshTimeout: 2000,
        dnsLookup: (address, callback) => callback(null, address),
        redisOptions: {
          username: redisConfig.username,
          password: redisConfig.password,
          db,
          tls: {}
        }
```
There's a PR on the frontend we need to merge

Reply:
Yeah but that is separate to the ESM change?

Reply:
Can we pick this up as a separate thing? As there is so much already in here anything extra would be good done in a different PR

Reply:
That is basically what the current code does atm, just slightly different conditional branching. Outcome is the same